src/chapter1/section1/Exercise33_Matrix_Operations.java

```diff
@@ -165,15 +164,15 @@ private static double dot(double[] x, double[] y) {
 	//Vector-matrix product
 	private static double[] mult(double[] y, double[][] a) {
 		
-		//Number of rows in matrix must be the same as the number of rows in vector
+		//Number of rows in matrix must be the same as the number of columns in vector
 		if (a == null || y == null || a.length != y.length) {
 			throw new IllegalArgumentException();
 		}
 		
 		double[] b = new double[a[0].length];
 		
-		for (int i = 0; i < a.length; i++) {
-			for (int j = 0; j < a[0].length; j++) {
+		for (int i = 0; i < a[0].length; i++) {
```

Comment:
Using `y.length` would make more sense here
```suggestion
		for (int i = 0; i < y.length; i++) {
```

Reply:
Yes you are right. I used this code only to show that we can be confused about the order of matrix and vector.